projects/plugins/jetpack/extensions/blocks/subscriptions/class-jetpack-subscription-site.php

```diff
@@ -39,6 +40,27 @@ public static function init() {
 	 */
 	public function handle_subscribe_block_placements() {
 		$this->handle_subscribe_block_post_end_placement();
+
+		add_filter(
+			'jetpack_options_whitelist',
+			function ( $options ) {
+				$options[] = 'jetpack_subscriptions_subscribe_navigation_enabled';
+
+				return $options;
+			}
+		);
+
+		// If called via REST API, we need to register later in the lifecycle
+		if ( ( new Host() )->is_wpcom_platform() && ! jetpack_is_frontend() ) {
+			add_action(
+				'restapi_theme_init',
+				function () {
+					Jetpack_Subscription_Site::init()->handle_subscribe_block_navigation_placement();
+				}
+			);
+		} else {
+			self::init()->handle_subscribe_block_navigation_placement();
+		}
```

Comment:
I think this is not needed at all. Just call `$this->handle_subscribe_block_navigation_placement();` instead and the late registering is already handled here:
https://github.com/Automattic/jetpack/blob/75b03102a816848ca9a6c0ca084887b6237a3f66/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php#L185-L195

The idea is the `handle_subscribe_block_placements` method should look like this:
```php
/**
 * Handles Subscribe block placements.
 *
 * @return void
 */
public function handle_subscribe_block_placements() {
	$this->handle_subscribe_block_post_end_placement();
	$this->handle_subscribe_block_navigation_placement();
}
```